lib/engine/game/base.rb

```diff
@@ -1512,7 +1512,7 @@ def upgrades_to_correct_color?(from, to, selected_company: nil)
 
       def upgrades_to_correct_label?(from, to)
         # If the from tile has a future label and the to tile is the color for it use that, otherwise use the from's label
-        return from.future_label.label == to.label.to_s if from.future_label && to.color == from.future_label.color
+        return from.future_label.label == to.label.to_s if from.future_label && to.color.to_s == from.future_label.color
```

Comment:
Why does `to` requires `to_s`, but not `from`?

Reply:
`from`is a `tile`, which (effectively) validates the string `color` when parsing the tile config and converts `color` to a `symbol` when the tile is being created, whereas in `future_label`, `color` is just a `string` causing this to `==` in JS, but `!=` in Ruby without the `to_s

```ruby
module Engine
  class Tile
    include Config::Tile

    attr_accessor :blocks_lay, :hex, :icons, :index, :legal_rotations, :location_name,
                  :name, :opposite, :reservations, :upgrades, :color, :future_label
    attr_reader :borders, :cities, :edges, :junction, :nodes, :labels, :parts, :preprinted, :rotation, :stops, :towns,
                :offboards, :blockers, :city_towns, :unlimited, :stubs, :partitions, :id, :frame, :stripes, :hidden

    ALL_EDGES = [0, 1, 2, 3, 4, 5].freeze

    def self.for(name, **opts)
      if (code = WHITE[name])
        color = :white
      elsif (code = YELLOW[name])
        color = :yellow
        ...
```